packages/ui/src/viewer-table/DataTable/SchemaTable/Columns/AllowedValuesColumn/ComputeAllowedValues.tsx

```diff
@@ -0,0 +1,372 @@
+			.map((mapping) => ({ foreignKey, mapping })),
+	);
+
+	const relevantForeignKeys = found?.map((item) => item.foreignKey);
+
+	const isBasicUniqueKey =
+		Array.isArray(uniqueKeys) && uniqueKeys.length === 1 && uniqueKeys[0] === currentSchemaField.name;
+
+	const isCompoundUniqueKey =
+		Array.isArray(uniqueKeys) && uniqueKeys.length > 1 && uniqueKeys.includes(currentSchemaField.name);
+
+	const compoundForeignKey = foreignKeys?.find(
+		(foreignKey) =>
+			foreignKey.mappings.length > 1 &&
+			foreignKey.mappings.some((mapping) => mapping.local === currentSchemaField.name),
+	);
+
+	const multipleSchemaReferences =
+		found && found.length > 1 ?
+			{
+				field: currentSchemaField.name,
+				associatedSchemas: relevantForeignKeys?.map((item) => item.schema),
+			}
+		:	undefined;
+
+	const computeRestrictions = [
+		{
+			condition: multipleSchemaReferences !== undefined,
+			content: (
+				<span>
+					Must reference an existing <FieldBlock>{multipleSchemaReferences?.field}</FieldBlock> within the{' '}
+					<b>
+						{multipleSchemaReferences?.associatedSchemas && multipleSchemaReferences.associatedSchemas.length > 1 ?
+							multipleSchemaReferences.associatedSchemas.slice(0, -1).join(', ') +
+							' and ' +
+							multipleSchemaReferences.associatedSchemas.slice(-1)
+						:	multipleSchemaReferences?.associatedSchemas?.[0]}
+					</b>{' '}
+					schemas.
+				</span>
+			),
+		},
+		{
+			condition: compoundForeignKey !== undefined,
+			content: (
+				<span>
+					Must reference an existing combination of:{' '}
+					<span
+						css={css`
+							display: inline-flex;
+							gap: 2px;
+						`}
+					>
+						{compoundForeignKey?.mappings.map((mapping) => (
+							<FieldBlock key={mapping.local}>{mapping.local}</FieldBlock>
+						))}
+					</span>{' '}
+					as defined in the <b>{compoundForeignKey?.schema}</b> schema.
+				</span>
+			),
+		},
+		{
+			condition: isBasicUniqueKey && relevantForeignKeys && relevantForeignKeys.length === 1,
+			content: (
+				<span>
+					Must reference an existing: <FieldBlock>{currentSchemaField.name}</FieldBlock> as defined in the{' '}
+					<b>{relevantForeignKeys?.[0]?.schema}</b> schema. Each record can only reference one{' '}
+					<b>{relevantForeignKeys?.[0]?.schema}</b>.
+				</span>
+			),
+		},
+		{
+			condition: !isBasicUniqueKey && relevantForeignKeys && relevantForeignKeys.length === 1,
+			content: (
+				<span>
+					Must reference an existing: <FieldBlock>{currentSchemaField.name}</FieldBlock> as defined in the{' '}
+					<b>{relevantForeignKeys?.[0]?.schema}</b> schema. Multiple records can reference the same{' '}
+					<b>{relevantForeignKeys?.[0]?.schema}</b>.
+				</span>
+			),
+		},
+		{
+			condition: isCompoundUniqueKey,
+			content: (
+				<span>
+					Must be unique in combination with:{' '}
+					{uniqueKeys
+						?.filter((key) => key !== currentSchemaField.name)
+						?.map((key) => <FieldBlock key={key}>{key}</FieldBlock>)}
```

Comment:
if you have a `filter` working followed by `map`.. there's always be at least an empty array so no need for `?`